src/test/java/com/example/rezerwowy/integration/TeamIT.java

```diff
@@ -0,0 +1,218 @@
+package com.example.rezerwowy.integration;
+
+import com.example.rezerwowy.dtos.TeamDto;
+import com.example.rezerwowy.models.Team;
+import com.example.rezerwowy.repositories.TeamRepository;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.annotation.DirtiesContext;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureTestDatabase
+class TeamIT {
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Autowired
+    private TeamRepository teamRepository;
+
+    @Test
+    @DirtiesContext
+    void should_createTeam_when_teamDoesntExist() {
+        // given
+        Team team = new Team(null, "FC Barcelona", "FCB");
+
+        // when
+        ResponseEntity<TeamDto> createResponse = restTemplate.postForEntity("/teams", team, TeamDto.class);
+
+        // then
+        Assertions.assertAll(
+                () -> assertThat(createResponse.getStatusCode()).isEqualTo(HttpStatus.CREATED),
+                () -> assertThat(teamRepository.existsById(createResponse.getBody().id())).isTrue(),
+                () -> assertThat(createResponse.getBody().id()).isNotNull(),
+                () -> assertThat(createResponse.getBody().name()).isEqualTo(team.getName()),
+                () -> assertThat(createResponse.getBody().abbreviation()).isEqualTo(team.getAbbreviation())
+        );
+    }
+
+    @Test
+    @DirtiesContext
+    void should_returnBadRequest_when_teamWithTheSameNameExists() {
+        // given
+        Team team = new Team(null, "FC Barcelona", "FCB");
+        Team team2 = new Team(null, "FC Barcelona", "FCB");
+
+        // when
+        ResponseEntity<TeamDto> createResponse = restTemplate.postForEntity("/teams", team, TeamDto.class);
```

Comment:
Znowu, testujesz 2 rzeczy na raz. Nie lepiej w warunkach początkowych po prostu to dodać do repozytorium?